presto-spi/src/main/java/com/facebook/presto/spi/type/SqlTime.java

```diff
@@ -17,32 +17,48 @@
 
 import java.time.Instant;
 import java.time.ZoneId;
+import java.time.ZoneOffset;
 import java.time.format.DateTimeFormatter;
 import java.util.Objects;
+import java.util.Optional;
 
 public final class SqlTime
 {
     private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm:ss.SSS");
 
     private final long millisUtc;
-    private final TimeZoneKey sessionTimeZoneKey;
+    private final Optional<TimeZoneKey> sessionTimeZoneKey;
 
+    public SqlTime(long millisUtc)
+    {
+        this.millisUtc = millisUtc;
+        this.sessionTimeZoneKey = Optional.empty();
+    }
+
+    @Deprecated
     public SqlTime(long millisUtc, TimeZoneKey sessionTimeZoneKey)
     {
         this.millisUtc = millisUtc;
-        this.sessionTimeZoneKey = sessionTimeZoneKey;
+        this.sessionTimeZoneKey = Optional.of(sessionTimeZoneKey);
     }
 
     public long getMillisUtc()
     {
         return millisUtc;
     }
 
-    public TimeZoneKey getSessionTimeZoneKey()
+    @Deprecated
+    public Optional<TimeZoneKey> getSessionTimeZoneKey()
```

Comment:
Depending on how you are using this. It may (and may not) be reasonable to do this instead:

```
public TimeZoneKey getSessionTimeZoneKey() {
  checkState(sessionTimeZoneKey.isPresent());
  return sessionTimeZoneKey.get();
}